src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/illegalidentifiername/Example1.java

```diff
@@ -0,0 +1,27 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="IllegalIdentifierName"/>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.illegalidentifiername;
+
+// xdoc section -- start
+public class Example1 {
+  public static void main(String... args) {
+    var var = 4; // violation
+    int record = 15; // violation
+    String yield = "yield"; // violation
+
+    record Record (Record record){} // 2 violations
+
+    String yieldString = "yieldString"; // ok, part of another word
+    record MyRecord(){} // ok, part of another word
+    var variable = 2; // ok, part of another word
+    String __; // ok, because underscore is part of another word
```

Comment:
lgtm....
I believe it doesn't throw a violation anymore.
This isn't ideal but let's wait for the opinion of maintainers.

Reply:
ok I will wait for the maintainers... thanks for the help

Reply:
If not compile able, remove it from example, it is trash. This is whole reason why we do this enablement as we have unrealistic java in examples.

Reply:
done